### PR TITLE
Fix synchronous functions being added to Asyncify.asyncExports

### PR DESCRIPTION
**Pull request by blexrob**
In https://github.com/emscripten-core/emscripten/commit/09e43acbc620d4d851dd883b980efe1cc96f7d17, the function `Asyncify.isAsyncExport` was added, which (according to the documentation) should return `true` for all functions that were wrapped with async WebAssembly.Functions.

However, in `instrumentWasmExports` (in library_async.js) all exports are added to this map, even those that do not match the `exportPatterns`.

This causes an error when `getWasmTableEntry` is called for that synchronous function, because that function will call `makeAsyncFunction` on that function if it's not present in the `wasmTableMirror` yet.

This can be reproduced by having an exported function of which the address is taken within the module, and calling addFunction. An exported function that has its address taken will be present in the `wasmTable`. `addFunction` will call `getFunctionAddress`, which will initialize the `functionsInTableMap` by calling `getWasmTableEntry` on all entries in the `wasmTable`. That function will call Asyncify.makeAsyncFunction on all functions for which Asyncify.isAsyncExport returns true. Because all exported functions (including synchronous ones) are added to the `Asyncify.asyncExports` set, this will fail on synchronous functions.

### CODE REVIEW

File: test/test_other.py [hunk 1]
```diff
@@ -2875,6 +2875,17 @@ def test_jspi_wildcard(self):
 
     self.do_runf(test_file('other/test_jspi_wildcard.c'), 'done')
 
+  def test_jspi_addfunction(self):
+    # make sure synchronous functions in the wasmTable aren't processed with Asyncify.makeAsyncFunction
+    self.require_v8()
```
Comment (brendandahl) [inline test/test_other.py:2880]: This can be `self.require_jspi()` and then you can get rid of v8 args below.

File: test/test_other.py [hunk 2]
```diff
@@ -2875,6 +2875,17 @@ def test_jspi_wildcard(self):
 
     self.do_runf(test_file('other/test_jspi_wildcard.c'), 'done')
 
+  def test_jspi_addfunction(self):
```
Comment (brendandahl) [inline test/test_other.py:2878]: Add underscore `test_jspi_add_function` and the filename too.

**Review by brendandahl — COMMENTED** [review]
Looks good, besides some nits. Thanks for fixing my bad merge!

File: test/test_other.py (hunk 2 above)
Comment (blexrob) [reply test/test_other.py]: done

File: test/test_other.py (hunk 1 above)
Comment (blexrob) [reply test/test_other.py]: done

**Review by brendandahl — APPROVED** [review]

**Pull request merged**